lib/xandra/cluster/control_connection.ex

```diff
@@ -445,11 +445,17 @@ defmodule Xandra.Cluster.ControlConnection do
               source: :xandra
             })
 
-            send(data.cluster, {:host_added, host})
-            Map.put(acc, peername, %{host: host, status: :up})
+            {existing_acc, Map.put(discovered_acc, peername, %{host: host, status: :up})}
```

Comment:
Actually, the order of `discovered_acc` is important, gonna change that to list.